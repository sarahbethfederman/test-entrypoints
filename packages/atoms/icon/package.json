{
  "name": "@auscred/icon",
  "version": "0.0.1",
  "source": "src/Icon.tsx",
  "main": "dist/Icon.js",
  "types": "dist/Icon.d.ts",
  "directories": {
    "lib": "lib"
  },
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "devDependencies": {
    "jest": "^23.1.0",
    "@lui/story-utils": "^1.0.0",
    "@storybook/addon-knobs": "^3.4.8",
    "@types/storybook__addon-knobs": "^3.3.1"
  },
  "scripts": {
    "build": "node ./src/build.js && tsc -p ./tsconfig.json"
  },
  "dependencies": {
    "svg-sprite": "^1.4.0"
  }
}
