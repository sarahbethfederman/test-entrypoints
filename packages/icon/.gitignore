src/icons-compiled/*
src/index.ts