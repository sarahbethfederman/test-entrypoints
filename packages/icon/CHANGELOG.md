## [9.6.2] - 2020-09-02 - [SUB-609][https://creditandfinance.atlassian.net/browse/sub-609]

## 10.0.0-beta.0

### Minor Changes

- preconstruct infra update

### Patch Changes

- Updated dependencies [undefined]
  - @lendi-ui/color@6.0.0-beta.0
  - @lendi-ui/theme@10.0.0-beta.0

### Added

- Added the domain bot Icon && Updated the autobot Icon name to LendiBotIcon

## [9.5.2] - 2020-08-27 - [SUB-601][https://creditandfinance.atlassian.net/browse/sub-601]

### Updated

- Updated Domain Home Loans for Team View

## [9.5.1] - 2020-08-18 - [SUB-461][https://creditandfinance.atlassian.net/browse/sub-461]

### Added

- Added Domain Home Loans logos

## [9.5.0] - 2020-04-28 - [SUB-242][https://creditandfinance.atlassian.net/browse/sub-242]

### Added

- Added autobot icon

## [9.4.1] - 2020-04-28 - [SUB-hotfix-unreadMessageClass]

### Added

- Updated the fillPath class on the unread message icon.

## [9.4.0] - 2020-04-28 - [SUB-hotfix-redDot]

### Added

- Added red dot icon.

## [9.3.0] - 2020-03-18 - [HUB-hotfix-domainIcon](https://creditandfinance.atlassian.net/browse/FUNNEL-1081)

### Added

- Added domain logomark icon.
- Added domain teamview logo icon.

## [9.2.0] - 2020-02-19 - [HUB-893](https://creditandfinance.atlassian.net/browse/HUB-893)

### Added

- Added unread icon.

## [9.1.1] - 2019-11-22

### Changed

- Fixed the colours in the new PropertyReport icon.

## [9.1.0] - 2019-11-22 - [HUB-664](https://creditandfinance.atlassian.net/browse/HUB-664)

### Added

- Added in relation to: HUB-664, HUB-665 and HUB-695
- Adds: ArrowRightCircle, BullseyeArrow, LightbulbOn, HomeMapMarker, HomeAccount, PropertyReport, CalculatorVariant, IcAssignment and CalendarMonth icons.

## [9.0.2] - 2019-09-27

- Change color to an optional property.

## [9.0.1] - 2019-09-26 [Release Notes](https://creditandfinance.atlassian.net/wiki/spaces/HUB/pages/803930391/Upcoming+Major+Changes)

- Bumped lendi-ui dependencies to latest versions

## [9.0.0] - 2019-9-23 [Release notes](https://creditandfinance.atlassian.net/wiki/spaces/HUB/pages/803930391/Upcoming+Major+Changes)

### Changed

- Uses Theme v7
- Update typescript to version 3.5.2 and react to version 16.8.4 [HUB-151](https://creditandfinance.atlassian.net/browse/HUB-151)

## [8.2.0] - 2019-07-16 - [HUB-330](https://creditandfinance.atlassian.net/browse/HUB-330)

### Changed

- Change icon default color to 'currentColor'

## [8.1.0] - 2019-06-07 - [HUB-302](https://creditandfinance.atlassian.net/browse/HUB-302)

### Changed

- Added the vertical-split icon

## [8.0.1] - 2019-06-07 - [HUB-264](https://creditandfinance.atlassian.net/browse/HUB-264)

### Changed

- Fix the icon alignment issue.

## [8.0.0] - 2019-05-29 - [HUB-144](https://creditandfinance.atlassian.net/browse/HUB-144)

### Deleted

- FileDownload, SuccessTwo, Warn, Error, Mobile (duplicate icons)
- All fancy icons (moved to the new fancy-icon package)

## Changed

- Renamed icons to match material.ui.

## [7.2.0] - 2019-05-17 - [HUB-174](https://creditandfinance.atlassian.net/browse/HUB-174)

### Changed

- Module format reverted back to CJS.

### Deleted

- Dropped "module" property from package.json.

## [7.1.2] - 2019-05-09 - [HUB-92](https://creditandfinance.atlassian.net/browse/HUB-92)

### Changed

- Updated CHANGELOG to match required parsing format.

## [7.1.1] - 2019-05-08 - [HUB-54](https://creditandfinance.atlassian.net/browse/HUB-54)

### Changed

- Change module main entry point to esm in package.json

## [7.1.0] - 2019-05-02 - [HUB-126](https://creditandfinance.atlassian.net/browse/HUB-126)

### Added

- Added the following new icons: file-upload & FileDownload as an alias to GetApp.

## [7.0.1] - 2019-05-02 - [HUB-132](https://creditandfinance.atlassian.net/browse/HUB-132)

### Changed

- Fix MS's Edge Icon rendering issue

## [7.0.0] - 2019-04-30 - [HUB-54](https://creditandfinance.atlassian.net/browse/HUB-54)

### Deleted

- Remove the CJS bundles.

## [6.0.0] - 2019-04-16 - [HUB-49](https://creditandfinance.atlassian.net/browse/HUB-49)

### Changed

- Remove Left and Right Icons.

## [5.3.1] - 2019-04-08 - [HUB-71](https://creditandfinance.atlassian.net/browse/HUB-71)

### Changed

- Added styling to the icon container to align text and icon baseline when using inline icons.

## [5.3.0] - 2019-04-01 - [HUB-53](https://creditandfinance.atlassian.net/browse/HUB-53)

### Added

- Added the following new icons: attach-file, assignment-returned, autorenew, assessment, save, credit-card, assignment-return, people, more-horiz, done-all, assignment-turned-in, error2, search, date-range, assignment, open-with, dashboard, timeline, gavel, trending-flat, report-problem, trending-down, delete, fullscreen, delete-outline, remove-circle, sentiment-satisfied-alt, launch, edit, add-alert, trending-up, help, history, schedule, room, notification-important, get-app, share, highlight-off, more-vert, favorite-border, hourglass-empty, info2, monetization-on, thumb-up, help-outline, error-outline, add, check2, warning, build, computer, data-usage, alarm, description, stay-current-portrait, power-settings-new, account-balance

## [5.2.0] - 2019-03-15 - [BRAVO-576](https://creditandfinance.atlassian.net/browse/BRAVO-576)

### Added

- Added un-circled warn, info and success icons `info-two success-two warn-two`

## [5.1.0] - 2019-03-06 - [BRAVO-572](https://creditandfinance.atlassian.net/browse/BRAVO-572)

### Added

- Added Maintenance icon and PageNotFound icon.

## [5.0.1] - 2019-03-05 - [BRAVO-387](https://creditandfinance.atlassian.net/browse/BRAVO-387)

### Added

- Added Domain logo.

## [5.0.0]

### Changed

- Breaking change version update

## [4.4.0]

### Changed

- Bump version for left and right icons
