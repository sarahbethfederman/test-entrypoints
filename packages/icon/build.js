const SVGSpriter = require('svg-sprite');
const fs = require('fs');
const path = require('path');
const mkdirp = require('mkdirp');

const directory = `${__dirname}/src/icons`;

const spritify = new SVGSpriter({
  dest: 'dist',
  mode: {
    symbol: true,
    inline: true,
  },
});

fs.readdir(directory, (err, files) => {
  if (err) {
    console.error(err);
    return;
  }

  files.forEach((file) => {
    if (file.endsWith('.svg')) {
      spritify.add(
        path.resolve(`${directory}/${file}`),
        file,
        fs.readFileSync(`${directory}/${file}`, { encoding: 'utf-8' })
      );
    }
  });

  spritify.compile(function(error, result, data) {
    for (var mode in result) {
      for (var resource in result[mode]) {
        fs.writeFileSync(`../../website/static/sprite.symbol.svg`, result[mode][resource].contents);
      }
    }
  });
});
