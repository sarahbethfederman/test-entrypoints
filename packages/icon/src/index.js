import React from 'react';

const Icon = () => <span>Icon</span>;

export default Icon;
