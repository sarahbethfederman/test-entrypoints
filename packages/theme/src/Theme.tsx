import * as React from 'react';
import styled, { ThemeProvider, injectGlobal } from 'styled-components';
import { theme } from './theme/lendi';
import { ThemeMap } from './types';

interface ThemeProps {
  kind?: ThemeMap;
  children?: React.ReactNode;
}

// Any cross browser resets should occur here
// tslint:disable-next-line no-unused-expression
injectGlobal`
  html, body {
    margin: 0;
    padding: 0;
  }

  :root {
    --lendi-ui-size: 14px;
  }

  @media (min-width: 36.0625em) {
    :root {
      --lendi-ui-size: 16px;
    }
  }
`;

const Root = styled.div`
  color: ${theme.colors.shade[700]};
`;

export function Theme(props: ThemeProps) {
  const { children } = props;
  const kind = props.kind || theme;

  return (
    <ThemeProvider theme={kind}>
      <Root>{children}</Root>
    </ThemeProvider>
  );
}
