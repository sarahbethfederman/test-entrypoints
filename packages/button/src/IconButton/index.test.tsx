import * as React from 'react';
import { mount } from 'enzyme';
import Theme from '@lendi-ui/theme';
import { Lock, Spinner } from '@lendi-ui/icon';
import { IconButton, IconButtonSize } from '.';

const sizes: IconButtonSize[] = ['lg', 'md', 'sm', 'xs'];

let button;

function render(props) {
  const element = mount(
    <Theme>
      <IconButton icon={Lock} {...props} />
    </Theme>
  );
  button = element.find(IconButton);
}

describe('IconButton', () => {
  it('should render the iconButton component with an icon', () => {
    render({});
    expect(button.find('svg')).toHaveLength(1);
  });

  sizes.forEach((size) => {
    it(`should render the "${size}" size correctly`, () => {
      render({ size });
      expect(button).toMatchSnapshot();
    });
  });

  it('should render an <a> tag when an href prop is provided', () => {
    render({ href: 'https://www.lendi.com.au/' });
    expect(button.find('a')).toHaveLength(1);
    expect(button.find('button')).toHaveLength(0);
  });

  it('should render a <button> when href is not specified', () => {
    const mockOnClick = jest.fn();
    render({ onClick: mockOnClick });
    expect(button.find('a')).toHaveLength(0);
    expect(button.find('button')).toHaveLength(1);
  });

  it("should render disabled style when the 'isDisabled' prop is true", () => {
    render({ href: 'https://www.lendi.com.au/', isDisabled: true });
    expect(button.find(IconButton)).toHaveStyleRule('cursor', 'not-allowed');
  });

  it("should render the Spinner icon when the 'isLoading' prop is true", () => {
    render({ isLoading: true });
    expect(button.find(Spinner)).toHaveLength(1);
  });

  it("should NOT call the 'onClick' function when the 'isDisabled' prop is true", () => {
    const mockOnClick = jest.fn();
    render({ href: 'https://www.lendi.com.au/', isDisabled: true, onClick: mockOnClick });
    button.find(IconButton).simulate('click');
    expect(mockOnClick).not.toBeCalled();
  });

  it('should pass aria label', () => {
    render({ 'aria-label': 'Locked' });
    expect(button.find('button[aria-label]').prop('aria-label')).toEqual('Locked');
  });

  it('should pass through any properties starting with "data-" to the icon-button', () => {
    const dataKey = 'data-test';
    const dataValue = '12345';
    render({ 'data-test': dataValue });
    expect(button.props()[dataKey]).toEqual(dataValue);
  });

  it('should not pass through unrelated properties', () => {
    const fakeKey = 'fake-data';
    render({ fakeKey: 'doesnt exist' });
    expect(button.props()[fakeKey]).toEqual(undefined);
  });

  it("should send back the onClick function with 'SyntheticEvent' event", () => {
    let myEvent = undefined;
    const onClick = (e) => (myEvent = e);
    render({ onClick });
    button.simulate('click');
    expect(myEvent.constructor.name === 'SyntheticEvent').toBeTruthy();
  });

  describe('test native props and Standard HTML Attributes', () => {
    it('should mount with Aria attributes', () => {
      const ARIA_LABEL = 'testLabel';
      const ARIA_DESCRIBE_BY = 'info';
      render({
        'aria-label': ARIA_LABEL,
        'aria-describedby': ARIA_DESCRIBE_BY,
      });
      expect(button.props()['aria-label']).toBe(ARIA_LABEL);
      expect(button.props()['aria-describedby']).toBe(ARIA_DESCRIBE_BY);
    });
    it('should mount with native props like id and title', () => {
      const TEXT_ID = 'testId';
      const TEXT_TITLE = 'testTitle';
      render({
        id: TEXT_ID,
        title: TEXT_TITLE,
      });
      expect(button.props()['id']).toBe(TEXT_ID);
      expect(button.props()['title']).toBe(TEXT_TITLE);
    });
  });
});
