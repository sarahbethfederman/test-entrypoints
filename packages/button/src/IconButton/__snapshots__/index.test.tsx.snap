// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should render the "lg" size correctly 1`] = `
.c1 {
  width: calc(2 * var(--lendi-ui-size));
  height: calc(2 * var(--lendi-ui-size));
  vertical-align: middle;
  display: inline-block;
}

.c1 > svg {
  fill: rgba(0,192,165,1);
  width: inherit;
  height: inherit;
}

.c1 .fillPath {
  fill: rgba(0,192,165,1);
}

.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  cursor: pointer;
  border: none;
  border-radius: 6px;
  text-align: center;
  -webkit-transition: all 0.1s;
  transition: all 0.1s;
  background: inherit;
  padding: var(--lendi-ui-size);
  background: inherit;
}

.c0:focus {
  background: none;
}

.c0:hover {
  background: inherit;
  background: rgba(74,74,74,0.15);
}

.c0:active {
  background: inherit;
  background: rgba(74,74,74,0.2);
}

<IconButton
  icon={[Function]}
  size="lg"
>
  <styled.button
    disabled={false}
    size="lg"
    type="button"
  >
    <StyledComponent
      disabled={false}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              "
  ",
              "
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  letter-spacing: normal;
",
              ";
  cursor: pointer;
  border: none;
  border-radius: ",
              [Function],
              ";
  text-align: center;
  transition: all 0.1s;
  background: inherit;

  :focus {
    background: none;
  }
  :hover {
    background: inherit; /* IE11 fallback */
    background: rgba(74, 74, 74, 0.15);
  }
  :active {
    background: inherit; /* IE11 fallback */
    background: rgba(74, 74, 74, 0.2);
  }
  ",
              [Function],
              "
  ",
              [Function],
              "
",
              ";
  background: inherit;
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      size="lg"
      type="button"
    >
      <button
        className="c0"
        disabled={false}
        size="lg"
        type="button"
      >
        <SvgLock
          color="primary.500"
          height="calc(2 * var(--lendi-ui-size))"
          width="calc(2 * var(--lendi-ui-size))"
        >
          <styled.span
            color="primary.500"
            height="calc(2 * var(--lendi-ui-size))"
            width="calc(2 * var(--lendi-ui-size))"
          >
            <StyledComponent
              color="primary.500"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  width: ",
                      [Function],
                      ";
  height: ",
                      [Function],
                      ";
  vertical-align: middle;
  display: inline-block;

  & > svg {
    fill: ",
                      [Function],
                      ";
    width: inherit;
    height: inherit;
  }

  .fillPath {
    fill: ",
                      [Function],
                      ";
  }
",
                    ],
                  },
                  "displayName": "styled.span",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "span",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              height="calc(2 * var(--lendi-ui-size))"
              width="calc(2 * var(--lendi-ui-size))"
            >
              <span
                className="c1"
                color="primary.500"
                height="calc(2 * var(--lendi-ui-size))"
                width="calc(2 * var(--lendi-ui-size))"
              >
                <svg
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                  />
                </svg>
              </span>
            </StyledComponent>
          </styled.span>
        </SvgLock>
      </button>
    </StyledComponent>
  </styled.button>
</IconButton>
`;

exports[`IconButton should render the "md" size correctly 1`] = `
.c1 {
  width: calc(1.5 * var(--lendi-ui-size));
  height: calc(1.5 * var(--lendi-ui-size));
  vertical-align: middle;
  display: inline-block;
}

.c1 > svg {
  fill: rgba(0,192,165,1);
  width: inherit;
  height: inherit;
}

.c1 .fillPath {
  fill: rgba(0,192,165,1);
}

.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  cursor: pointer;
  border: none;
  border-radius: 6px;
  text-align: center;
  -webkit-transition: all 0.1s;
  transition: all 0.1s;
  background: inherit;
  padding: calc(0.75 * var(--lendi-ui-size));
  background: inherit;
}

.c0:focus {
  background: none;
}

.c0:hover {
  background: inherit;
  background: rgba(74,74,74,0.15);
}

.c0:active {
  background: inherit;
  background: rgba(74,74,74,0.2);
}

<IconButton
  icon={[Function]}
  size="md"
>
  <styled.button
    disabled={false}
    size="md"
    type="button"
  >
    <StyledComponent
      disabled={false}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              "
  ",
              "
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  letter-spacing: normal;
",
              ";
  cursor: pointer;
  border: none;
  border-radius: ",
              [Function],
              ";
  text-align: center;
  transition: all 0.1s;
  background: inherit;

  :focus {
    background: none;
  }
  :hover {
    background: inherit; /* IE11 fallback */
    background: rgba(74, 74, 74, 0.15);
  }
  :active {
    background: inherit; /* IE11 fallback */
    background: rgba(74, 74, 74, 0.2);
  }
  ",
              [Function],
              "
  ",
              [Function],
              "
",
              ";
  background: inherit;
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      size="md"
      type="button"
    >
      <button
        className="c0"
        disabled={false}
        size="md"
        type="button"
      >
        <SvgLock
          color="primary.500"
          height="calc(1.5 * var(--lendi-ui-size))"
          width="calc(1.5 * var(--lendi-ui-size))"
        >
          <styled.span
            color="primary.500"
            height="calc(1.5 * var(--lendi-ui-size))"
            width="calc(1.5 * var(--lendi-ui-size))"
          >
            <StyledComponent
              color="primary.500"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  width: ",
                      [Function],
                      ";
  height: ",
                      [Function],
                      ";
  vertical-align: middle;
  display: inline-block;

  & > svg {
    fill: ",
                      [Function],
                      ";
    width: inherit;
    height: inherit;
  }

  .fillPath {
    fill: ",
                      [Function],
                      ";
  }
",
                    ],
                  },
                  "displayName": "styled.span",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "span",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              height="calc(1.5 * var(--lendi-ui-size))"
              width="calc(1.5 * var(--lendi-ui-size))"
            >
              <span
                className="c1"
                color="primary.500"
                height="calc(1.5 * var(--lendi-ui-size))"
                width="calc(1.5 * var(--lendi-ui-size))"
              >
                <svg
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                  />
                </svg>
              </span>
            </StyledComponent>
          </styled.span>
        </SvgLock>
      </button>
    </StyledComponent>
  </styled.button>
</IconButton>
`;

exports[`IconButton should render the "sm" size correctly 1`] = `
.c1 {
  width: calc(1.25 * var(--lendi-ui-size));
  height: calc(1.25 * var(--lendi-ui-size));
  vertical-align: middle;
  display: inline-block;
}

.c1 > svg {
  fill: rgba(0,192,165,1);
  width: inherit;
  height: inherit;
}

.c1 .fillPath {
  fill: rgba(0,192,165,1);
}

.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  cursor: pointer;
  border: none;
  border-radius: 6px;
  text-align: center;
  -webkit-transition: all 0.1s;
  transition: all 0.1s;
  background: inherit;
  padding: calc(0.625 * var(--lendi-ui-size));
  background: inherit;
}

.c0:focus {
  background: none;
}

.c0:hover {
  background: inherit;
  background: rgba(74,74,74,0.15);
}

.c0:active {
  background: inherit;
  background: rgba(74,74,74,0.2);
}

<IconButton
  icon={[Function]}
  size="sm"
>
  <styled.button
    disabled={false}
    size="sm"
    type="button"
  >
    <StyledComponent
      disabled={false}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              "
  ",
              "
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  letter-spacing: normal;
",
              ";
  cursor: pointer;
  border: none;
  border-radius: ",
              [Function],
              ";
  text-align: center;
  transition: all 0.1s;
  background: inherit;

  :focus {
    background: none;
  }
  :hover {
    background: inherit; /* IE11 fallback */
    background: rgba(74, 74, 74, 0.15);
  }
  :active {
    background: inherit; /* IE11 fallback */
    background: rgba(74, 74, 74, 0.2);
  }
  ",
              [Function],
              "
  ",
              [Function],
              "
",
              ";
  background: inherit;
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      size="sm"
      type="button"
    >
      <button
        className="c0"
        disabled={false}
        size="sm"
        type="button"
      >
        <SvgLock
          color="primary.500"
          height="calc(1.25 * var(--lendi-ui-size))"
          width="calc(1.25 * var(--lendi-ui-size))"
        >
          <styled.span
            color="primary.500"
            height="calc(1.25 * var(--lendi-ui-size))"
            width="calc(1.25 * var(--lendi-ui-size))"
          >
            <StyledComponent
              color="primary.500"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  width: ",
                      [Function],
                      ";
  height: ",
                      [Function],
                      ";
  vertical-align: middle;
  display: inline-block;

  & > svg {
    fill: ",
                      [Function],
                      ";
    width: inherit;
    height: inherit;
  }

  .fillPath {
    fill: ",
                      [Function],
                      ";
  }
",
                    ],
                  },
                  "displayName": "styled.span",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "span",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              height="calc(1.25 * var(--lendi-ui-size))"
              width="calc(1.25 * var(--lendi-ui-size))"
            >
              <span
                className="c1"
                color="primary.500"
                height="calc(1.25 * var(--lendi-ui-size))"
                width="calc(1.25 * var(--lendi-ui-size))"
              >
                <svg
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                  />
                </svg>
              </span>
            </StyledComponent>
          </styled.span>
        </SvgLock>
      </button>
    </StyledComponent>
  </styled.button>
</IconButton>
`;

exports[`IconButton should render the "xs" size correctly 1`] = `
.c1 {
  width: var(--lendi-ui-size);
  height: var(--lendi-ui-size);
  vertical-align: middle;
  display: inline-block;
}

.c1 > svg {
  fill: rgba(0,192,165,1);
  width: inherit;
  height: inherit;
}

.c1 .fillPath {
  fill: rgba(0,192,165,1);
}

.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  cursor: pointer;
  border: none;
  border-radius: 6px;
  text-align: center;
  -webkit-transition: all 0.1s;
  transition: all 0.1s;
  background: inherit;
  padding: calc(0.375 * var(--lendi-ui-size));
  background: inherit;
}

.c0:focus {
  background: none;
}

.c0:hover {
  background: inherit;
  background: rgba(74,74,74,0.15);
}

.c0:active {
  background: inherit;
  background: rgba(74,74,74,0.2);
}

<IconButton
  icon={[Function]}
  size="xs"
>
  <styled.button
    disabled={false}
    size="xs"
    type="button"
  >
    <StyledComponent
      disabled={false}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              "
  ",
              "
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  letter-spacing: normal;
",
              ";
  cursor: pointer;
  border: none;
  border-radius: ",
              [Function],
              ";
  text-align: center;
  transition: all 0.1s;
  background: inherit;

  :focus {
    background: none;
  }
  :hover {
    background: inherit; /* IE11 fallback */
    background: rgba(74, 74, 74, 0.15);
  }
  :active {
    background: inherit; /* IE11 fallback */
    background: rgba(74, 74, 74, 0.2);
  }
  ",
              [Function],
              "
  ",
              [Function],
              "
",
              ";
  background: inherit;
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      size="xs"
      type="button"
    >
      <button
        className="c0"
        disabled={false}
        size="xs"
        type="button"
      >
        <SvgLock
          color="primary.500"
          height="var(--lendi-ui-size)"
          width="var(--lendi-ui-size)"
        >
          <styled.span
            color="primary.500"
            height="var(--lendi-ui-size)"
            width="var(--lendi-ui-size)"
          >
            <StyledComponent
              color="primary.500"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  width: ",
                      [Function],
                      ";
  height: ",
                      [Function],
                      ";
  vertical-align: middle;
  display: inline-block;

  & > svg {
    fill: ",
                      [Function],
                      ";
    width: inherit;
    height: inherit;
  }

  .fillPath {
    fill: ",
                      [Function],
                      ";
  }
",
                    ],
                  },
                  "displayName": "styled.span",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "span",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              height="var(--lendi-ui-size)"
              width="var(--lendi-ui-size)"
            >
              <span
                className="c1"
                color="primary.500"
                height="var(--lendi-ui-size)"
                width="var(--lendi-ui-size)"
              >
                <svg
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                  />
                </svg>
              </span>
            </StyledComponent>
          </styled.span>
        </SvgLock>
      </button>
    </StyledComponent>
  </styled.button>
</IconButton>
`;
